components/compliance-service/ingest/ingestic/ingestic.go

```diff
@@ -854,43 +854,38 @@ func convertProjectTaggingRulesToEsParams(projectTaggingRules map[string]*authz.
 	return map[string]interface{}{"projects": esProjectCollection}
 }
 
-func (backend *ESClient) GetDocByReportUUId(ctx context.Context, data *relaxting.ESInSpecReport, index string) (*relaxting.ESInSpecReport, error) {
+func (backend *ESClient) GetDocByReportUUId(ctx context.Context, reportUuid string, index string) (*relaxting.ESInSpecReport, error) {
 	logrus.Debug("Fetching project by UUID")
-	var item relaxting.ESInSpecReport
 
-	fsc := elastic.NewFetchSourceContext(false)
-	termQueryNotThisReport := elastic.NewTermsQuery("_id", data.ReportID)
+	var item relaxting.ESInSpecReport
 	boolQuery := elastic.NewBoolQuery()
-	boolQuery = boolQuery.Must(termQueryNotThisReport)
+
+	idsQuery := elastic.NewIdsQuery()
+	idsQuery.Ids(reportUuid)
+	boolQuery = boolQuery.Must(idsQuery)
+	fsc := elastic.NewFetchSourceContext(true)
 	searchSource := elastic.NewSearchSource().
 		FetchSourceContext(fsc).
 		Query(boolQuery).
-		Size(1000)
+		Size(1)
+
 	searchResult, err := backend.client.Search().
 		SearchSource(searchSource).
 		Index(index).
-		Do(context.Background())
+		FilterPath(
+			"took",
+			"hits.total",
+			"hits.hits._id",
+			"hits.hits._source",
+			"hits.hits.inner_hits").
+		Do(ctx)
 
 	if err != nil {
-		switch {
```

Comment:
Why did we remove this?

Reply:
It was not returning the correct values, and error will not come in the conditions which was there @kalroy